tracer/src/Datadog.Trace/Logging/TracerFlare/DebugLogReader.cs

```diff
@@ -0,0 +1,94 @@
+﻿// <copyright file="DebugLogReader.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+#nullable enable
+
+using System;
+using System.IO;
+using System.IO.Compression;
+using System.Threading.Tasks;
+
+namespace Datadog.Trace.Logging.TracerFlare;
+
+internal static class DebugLogReader
+{
+    private static readonly IDatadogLogger Log = DatadogLogging.GetLoggerFor(typeof(DebugLogReader));
+
+    public static bool TryToCreateSentinelFile(string logDirectory, string flareRequestId)
+    {
+        // Assuming the log directory is accessible and writeable
+        // If it isn't this will return false anyway, but we do an extra check, just to be sure
+        try
+        {
+            // Using the dotnet-tracer-sentinel-*.log pattern means it gets automatically cleaned up by the log cleaning task
+            var filePath = Path.Combine(logDirectory, $"dotnet-tracer-sentinel-{flareRequestId}.log");
+            using var file = File.Open(filePath, FileMode.CreateNew, FileAccess.Write);
+
+            // if we got this far, the file was created
+            return true;
```

Comment:
Can be left for a future PR, but a comment in the RFC indicated that NFS doesn't have exclusive file access. I'm not 100% clear on the implications, but hypothetically it means that this condition can succeed for multiple tracers. 

I wonder if it would be safer to:
 - Create the file with a different name
 - Write the runtime id in it
 - Rename the file. If it fails then the tracer lost the race
 - Read the runtime id from the file, if it's different then the tracer lost the race

Maybe it can be much simpler, but without a NFS system to test I can only make assumptions (for instance the renaming step is maybe enough).


Anyway, just leaving this as a note for now. I need to figure out a way to test on NFS.

Reply:
Yeah, I guess we can leave this for later as the logic is self contained here. Also, it's annoying but not _problematic_ that multiple tracers might upload the same files I guess?

Reply:
True